### PR TITLE
[fuchsia][flatland] Fix for AwaitVsync race

### PR DESCRIPTION
**Pull request by freiling**
Fixes some racy behavior in FlatlandConnection. Regression tests included in change

This PR includes a number of changes from other authors that havent landed yet. Once those changes land I will rebase this PR on top of them.

### CODE REVIEW

**Comment by freiling** [comment]
@uysalere I couldnt add you as a reviewer i dont know why

**Review by arbreng — APPROVED** [review]
Fantastic!  This is exactly the fix I was trying to describe the other day

Merge away once https://github.com/flutter/engine/pull/28850 is merged

**Comment by arbreng** [comment]
> @uysalere I couldnt add you as a reviewer i dont know why

It's b/c people have to go through https://github.com/flutter/flutter/wiki/Contributor-access before you can add them as a reviewer.  Not worth it for occasional contributors (and we will be leaving this repo soon anyways)

File: shell/platform/fuchsia/flutter/flatland_connection.cc [hunk 1]
```diff
@@ -51,18 +51,36 @@ void FlatlandConnection::Present() {
...
 
+// Note that this method is called from the UI thread.
 void FlatlandConnection::AwaitVsyncForSecondaryCallback(
-    FireCallbackCallback callback) {}
+    FireCallbackCallback callback) {
+  fml::TimePoint now = fml::TimePoint::Now();
+  callback(now, now);
+}
 
 void FlatlandConnection::OnError(
```
Comment (uysalere) [inline shell/platform/fuchsia/flutter/flatland_connection.cc:85]: Above comments on which thread methods runs is very useful? Can we add the rest too, i.e. raster thread?

**Review by uysalere — APPROVED** [review]
Looks good!

File: shell/platform/fuchsia/flutter/flatland_connection.cc (hunk 1 above)
Comment (freiling) [reply shell/platform/fuchsia/flutter/flatland_connection.cc:91]: Done

**Pull request merged**